src/PanAndZoomContainer.js

```diff
@@ -59,17 +54,44 @@ const PanAndZoomContainer = ({
   );
 
   useEffect(() => {
-    diagramContainerRef.current.style.transform = transform;
-  }, [transform]);
-
-  useLayoutEffect(() => {
-    containerRef.current.scrollBy({
-      left:
-        (scroll.left * (zoom - previousZoom.current)) / previousZoom.current,
-      top: (scroll.top * (zoom - previousZoom.current)) / previousZoom.current
-    });
+    const container = containerRef.current;
+
+    const containerWidth = container.clientWidth;
+    const containerHeight = container.clientHeight;
+
+    let x = pan.x,
+      y = pan.y;
+
+    const extremeX = Math.max(contentSpan.x, containerWidth),
+      extremeY = Math.max(contentSpan.y, containerHeight);
+
+    if (containerWidth >= extremeX * zoom) {
+      x = ((containerWidth - extremeX) * zoom) / 2;
+    }
+    if (containerHeight >= extremeY * zoom) {
+      y = ((containerHeight - extremeY) * zoom) / 2;
+    }
+
+    diagramContainerRef.current.style.transform = `translate3D(${x}px, ${y}px, 0) scale(${zoom})`;
+
+    container.scrollLeft = containerRef.current.scrollLeft =
+      (scroll.left * zoom) / previousZoom.current +
+      ((zoom - previousZoom.current) * containerWidth) /
+        (2 * previousZoom.current);
+    container.scrollTop =
```

Comment:
`containerRef.current.scrollTop` not updated here? As done on [line 77](https://github.com/ujjawalpabreja99/react-virtualized-flowchart/pull/2/files#diff-fa50ff659ac0d0bad829809f3061f7a785c9852fd029b4dacad4215138058323R77)

Reply:
line 77 was a typo updated now.